tools/clang/lib/AST/DeclPrinter.cpp

```diff
@@ -330,9 +330,8 @@ void DeclPrinter::VisitDeclContext(DeclContext *DC, bool Indent) {
       continue;
     }
 
-    if (isa<AccessSpecDecl>(*D)
-        && !Policy.LangOpts.HLSL // HLSL Change - no access specifier for hlsl.
-        ) {
+    if (isa<AccessSpecDecl>(*D))
+      if (!Policy.LangOpts.HLSL) { // HLSL Change - no access specifier for hlsl.
```

Comment:
I'm curious why this change was necessary.
Is there anything wrong with doing

```cpp
if (isa<AccessSpecDecl>(*D) && !Policy.LangOpts.HLSL)
```
?

Reply:
This was clang being. . . let's say extra cautious and maybe too clever by half. Clang realizes that Policy.LangOpts.HLSL is always true in this build. Consequently the overall expression always fails regardless of the return value of isa<AccessSpecDecl>() So the return value doesn't matter so it's kind of ignored, which there is an attribute added to it to make sure never happens.

The attribute is surely intended to make sure that it isn't called as a standalone function because its return value is its only effect and such a usage would be a problem. I think the clang warning is useful, this is a weird case where it is oversensitive, but it was easier to humor it than any alternative.

Reply:
I see. Thanks for clarifying.